Added parsing support for failed wordle games

stevenjmaev opened the pull request:
Resolves #2.
* Instead of having a `/(\d)/` as the regex for parsing the score, changed it to `/(.)/` to account for any character as well. 
  * As of 2022-02-11, the character that is given as a score for someone who failed to guess the wordle is `'X'`. 

* If the result of `parseInt()` is `NaN` then it most likely means that the "score" is `'X'`, in which case the `grid_height` is set to the max (6), the score is set to 1 above the max score, and `failed` is set to `true`.
  * I chose `failed` because `win` is a little misleading, since they didn't get the _best score_, they merely _didn't fail_.

* The text is sliced according to this new variable `grid_height` rather than `score`

stevenjmaev commented:
NOTE: NOT READY FOR MERGE!! NEW BUGS APPEARED!!! 😢

hkennyv commented:
> I chose failed because win is a little misleading, since they didn't get the best score, they merely didn't fail.

Love this attention to detail, excellent choice of positive wording!

stevenjmaev commented:
Ready for merge!! I added a new functionality to the debug mode so that if `DEBUG==1`, then we could post messages today as if it were yesterday. 

Also, I changed [this following code section](https://github.com/hkennyv/wordle-slack-bot/blob/eadd4990fffa96f4c59ce45f3f192ef019c14739/src/app.js#L80-L95) so that it checks the number of submissions rather than the number of winners (e.g. if only _one person_ made a submission and it was a failed submission, that user isn't classified as a winner, so `winners.length` is `0`).

Future work: that same code snippet could be restructured as a `switch` or nested `if-else` statement, but I think it's okay for now

hkennyv commented:
@stevenjmaev magnifiqué! merging

stevenjmaev commented on src/app.js:
```diff
@@ -47,20 +47,30 @@ const submissions = [];
+  if (isNaN(score)){ 
+    grid_height = 6;
+    failed = true;
+    score = grid_height + 1
```
@hkennyv yes that's correct. For now it's good, but in the future if the NYT ever decides to change the number of guesses to say 5, or 7. Then we'll have to add extra parsing
